src/autorest-core/lib/pipeline/plugins/autorest-dotnet.ts

```diff
@@ -3,10 +3,125 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { Lazy } from "../../lazy";
+import { AutoRestConfigurationImpl } from "../../configuration";
+import { ChildProcess } from "child_process";
+import { EventEmitter } from "../../events";
+import { CancellationToken } from "../../ref/cancallation";
 import { LazyPromise } from '../../lazy';
 import { SpawnJsonRpcAutoRest } from "../../../interop/autorest-dotnet";
 import { AutoRestPlugin } from "../plugin-endpoint";
+import { DataHandleRead, DataStoreViewReadonly, QuickScope, DataStoreView } from "../../data-store/data-store";
+import { Message } from "../../message";
+import { ConfigurationView } from "../../autorest-core";
 
+export class AutoRestDotNetPlugin extends EventEmitter {
+  private static instance = new Lazy<AutoRestDotNetPlugin>(() => new AutoRestDotNetPlugin());
+  public static Get(): AutoRestDotNetPlugin { return AutoRestDotNetPlugin.instance.Value; }
+
+  private childProc: ChildProcess;
+  public pluginEndpoint: Promise<AutoRestPlugin>;
+
+  private constructor() {
+    super();
+    this.childProc = SpawnJsonRpcAutoRest();
+    this.pluginEndpoint = AutoRestPlugin.FromChildProcess(this.childProc);
+  }
+
+  /**
+   * Probes whether the extension even supports the requested plugin.
+   */
+  public async CautiousProcess(
+    pluginName: string,
+    configuration: (key: string) => any,
+    inputScope: DataStoreViewReadonly,
+    outputScope: DataStoreView,
+    onMessage: (message: Message) => void): Promise<void> {
+
+    const ep = await this.pluginEndpoint;
+
+    // probe
+    const pluginNames = await ep.GetPluginNames(CancellationToken.None);
+    if (pluginNames.indexOf(pluginName) === -1) {
+      throw new Error(`The AutoRest dotnet extension does not offer a plugin called '${pluginName}'.`);
+    }
+    // process
+    const success = await ep.Process(pluginName, configuration, inputScope, outputScope, onMessage, CancellationToken.None);
+    if (!success) {
+      throw new Error(`Plugin ${pluginName} failed.`);
+    }
+  }
+
+  async Validate(swagger: DataHandleRead, workingScope: DataStoreView, onMessage: (message: Message) => void, isIndividual: boolean, openapiType: string): Promise<void> {
```

Comment:
not sure I understand correctly, but 99% "at the AzureValidator endpoint in validator itself" is the right answer. The core (this right here) shouldn't bother too much about details like that and just "forward" as many settings as possible. That allows updates in the validation space without changing the core.